### ISSUE TITLE
Feature Request: Ability to override save/restore for refresh token

### ISSUE THREAD

**Issue by mehcode** [algesten/hangupsjs#30]
I'd like to be able to override these two methods: https://github.com/algesten/hangupsjs/blob/master/src/auth.coffee#L103-L117

Either letting me passing in a derived Auth class or an object with load/store that Auth calls. Something like that. Again, if you're amicable I don't mind implementing.

**Comment by algesten** [comment]
go for it!

**Comment by algesten** [comment]
my preferred method would be hm. an `opts.tokenPersistence` which is any plain object with two functions:

```coffee
opts.tokenPersistence = {
  save:(token) -> ... returns promise 
  load:-> ... returns promise 
}
```

**Comment by algesten** [comment]
closed by 66b771e9f25126af3f2692bfd617882d726c30b9
